File: Homework1/src/Zpm.java
/*
 * Zpm.java
 * This class contains the main method to execute the Zpm program.
 * Zpm is a command-line tool for running scripts written in a custom scripting language.
 * 
 * Copyright (c) 2024 Ryan Van Witzenburg and ChatGPT. All rights reserved.
 */

/**
 * The Zpm class contains the main method to execute the Zpm program.
 * Zpm is a command-line tool for running scripts written in a custom scripting language.
 */
public class Zpm {

    /**
     * The main method of the Zpm program.
     * It takes a single command-line argument representing the filename of the script to be executed.
     * If the correct number of arguments is not provided, it prints an error message and exits.
     * Otherwise, it creates an instance of the Parser class, runs the script, and executes it.
     *
     * @param args the command-line arguments
     */
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Need to add java Zpm file to run statement");
            return;
        }

        String fileName = args[0]; // Get the filename from command-line arguments
        Parser parser = new Parser(); // Create an instance of the Parser class
        parser.run(fileName); // Run the script specified by the filename
    }

}

File: Homework6/Zpm.cpp
// Copyright 2024 Ryan Van Witzenburg and ChatGPT
#include <iostream>
#include "Parser.h"

using namespace std;

int main(int argc, char *argv[])
{
    if (argc != 2)
    {
        cout << "Need to add C++ Zpm file to run statement" << endl;
        return 1;
    }

    string fileName = argv[1];
    Parser parser;
    parser.run(fileName);

    return 0;
}
